### PR TITLE
DINT-63: implement initiate-checkout event for snowplow + aep

### PR DESCRIPTION
**Pull request by aniham**

### CODE REVIEW

File: src/handlers/shoppingCart/initiateCheckout.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    trackStructEvent({
+        category: "shopping-cart",
+        action: "initiate-checkout",
+        property: pageContext?.pageType,
+        context,
+    });
+};
+
+export default handler;
```
Comment (grahamcrackers) [inline src/handlers/shoppingCart/initiateCheckout.ts:27]: This is find for now, but i'm going to suggest that we eventually do away with default exports. It makes it harder for typescript to pick up functions in the IDE and adds a layer of obfuscation that really isn't neccessary. Just wanted to put it on everyone's radar.

File: src/handlers/shoppingCart/initiateCheckoutAEP.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                cartId: shoppingCartContext.id ?? "1",
+            },
+        },
+        productListItems: createProductListItems(shoppingCartContext),
+    };
+
+    alloy("sendEvent", { xdm: payload });
+};
+
+export default handler;
```
Comment (grahamcrackers) [inline src/handlers/shoppingCart/initiateCheckoutAEP.ts:35]: same as other default export

**Review by grahamcrackers — APPROVED** [review]

File: src/handlers/shoppingCart/initiateCheckout.ts (hunk 1 above)
Comment (aniham) [reply src/handlers/shoppingCart/initiateCheckout.ts:27]: @grahamcrackers i passionately hate the default exports - like i always say, i'm trying to be as undisruptive as possible but i 100% agree - track in new jira story?

File: src/handlers/shoppingCart/initiateCheckoutAEP.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+const handler = async (event: Event): Promise<void> => {
+    const alloy = await getAlloy();
+    const { shoppingCartContext } = event.eventInfo;
+    const shoppingCartCtx = createShoppingCartCtx(shoppingCartContext);
+
+    const context: Array<SelfDescribingJson> = [];
+
+    if (shoppingCartCtx) {
+        context.push(shoppingCartCtx);
```
Comment (jekelly-adobe) [inline src/handlers/shoppingCart/initiateCheckoutAEP.ts:19]: `context` is unused

File: src/handlers/shoppingCart/initiateCheckout.ts (hunk 1 above)
Comment (grahamcrackers) [reply src/handlers/shoppingCart/initiateCheckout.ts:27]: 💯 00000000 yes

File: src/handlers/shoppingCart/initiateCheckoutAEP.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import { createShoppingCartCtx } from "../../contexts";
+import { BeaconSchema } from "../../types/aep";
+import { createProductListItems } from "../../utils/aep/productListItems";
+
+const XDM_EVENT_TYPE = "commerce.checkouts";
+
+const handler = async (event: Event): Promise<void> => {
+    const alloy = await getAlloy();
+    const { shoppingCartContext } = event.eventInfo;
+    const shoppingCartCtx = createShoppingCartCtx(shoppingCartContext);
```
Comment (jekelly-adobe) [inline src/handlers/shoppingCart/initiateCheckoutAEP.ts:14]: `shoppingCartCtx` is not needed once the unused `context` is removed (see comment further below)

**Comment by jekelly-adobe** [comment]
@aniham still have unused stuff in `src/handlers/shoppingCart/initiateCheckoutAEP.ts` (see github warnings)

**Pull request merged**